app/team.go

```diff
@@ -575,12 +592,21 @@ func (a *App) AddTeamMember(teamId, userId string) (*model.TeamMember, *model.Ap
 		return nil, err
 	}
 
-	var teamMember *model.TeamMember
-	var err *model.AppError
-	if teamMember, err = a.GetTeamMember(teamId, userId); err != nil {
+	teamMember, err := a.GetTeamMember(teamId, userId)
+	if err != nil {
 		return nil, err
 	}
 
+	if a.PluginsReady() {
+		a.Go(func() {
+			pluginContext := &plugin.Context{}
+			a.Plugins.RunMultiPluginHook(func(hooks plugin.Hooks) bool {
+				hooks.UserHasJoinedTeam(pluginContext, teamMember, nil)
+				return true
+			}, plugin.UserHasJoinedTeamId)
+		})
+	}
```

Comment:
Similar comment here (and actually I think this one calls `JoinUserToTeam` so this hook might be getting triggered twice and you might just be able to remove this one and it'll work

Reply:
Yup, just having it in `JoinUserToTeam` seems to be sufficient.